hypothesis-python/tests/array_api/test_arrays.py

```diff
@@ -469,22 +481,21 @@ def test_may_reuse_distinct_integers_if_asked():
     )
 
 
-@pytest.mark.skipif(
-    not WIDTHS_FTZ[32], reason="Subnormals are valid for non-FTZ builds"
-)
-def test_cannot_draw_subnormals_for_ftz_float32():
+def test_cannot_draw_subnormals_for_ftz_float32(xp, xps):
     """For FTZ builds of array modules, strategy with subnormal elements
     strategy raises helpful error."""
-    strat = xps.arrays(
-        xp.float32,
-        10,
-        elements={
-            "min_value": 0.0,
-            "max_value": width_smallest_normals[32],
-            "exclude_min": True,
-            "exclude_max": True,
-            "allow_subnormal": True,
-        },
-    )
+    if not flushes_to_zero(xp, width=32):
+        pytest.skip("Subnormals are valid for non-FTZ builds")
```

Comment:
Does this make more sense?
```suggestion
    if not flushes_to_zero(xp, width=32):
        pytest.skip(
            f"Build of module {xp.__name__} does NOT flush subnormals to zero "
            "for float32, so xps.arrays() will not raise an error here (which "
            "is what we want to test)"
        )
```
Also in `test_from_dtypes.py` I could do with elaborating on the skip reason for two similar FTZ tests.

Reply:
OK, I think the way I'd want to write this one is 

```python
strat = xps.arrays(xp.float32, 10, elements=elements)
if flushes_to_zero(xp, width=32):
    with pytest.raises(InvalidArgument, match="Generated subnormal float"):
        strat.example()
else:
    strat.example()
```

Immediately obvious on reading this that we expect an error if it flushes to zero, but not otherwise 🙂

Reply:
Makes sense! Made this change and renamed this test case `test_subnormal_elements_validation`. Also did something similar in `test_from_dtype.py` in `test_subnormal_generation`.